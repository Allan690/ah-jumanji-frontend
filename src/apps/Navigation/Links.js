import React from "react";

import NavItem from "./NavItem";
import NavButton from "./NavButton";
import DropDownItem from './NavDropDown';
import DropDownNotification from './NavDropDownNotification';
import defaultUserIcon from '../../assets/img/default-avatar.jpg';


const NotificationLink = () => (
  <React.Fragment>
    <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
      <span className="badge badge-danger notifications-badge">
        2
      </span>
      <i className="fas fa-bell" />
      Notifications
    </button>
  </React.Fragment>
);

export const HomeLinks = () => (
  <React.Fragment>
    <NavItem classnameOuter="nav-link active" classname="fas fa-home" link="/a/home" label="Home" />
    <NavItem classnameOuter="nav-link" classname="fas fa-search" data-toggle="modal" data-target="#search" link="javascript:;" label="Search" />
    <li className="nav-item dropdown">
      <NotificationLink />
      <ul className="dropdown-menu" id="dropDownNotifications" aria-labelledby="navbarDropdown">
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
      </ul>
    </li>
    <li className="nav-item dropdown">
      <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <img src={defaultUserIcon} alt="follower avatar" className="followers-avatar navbar-avatar rounded-circle" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
    <NavButton classnameOuter="nav-link" classname="btn btn-success btn-sm fas fa-plus-square" link="/a/createarticle" label="&nbsp;Add Post" />
  </React.Fragment>
);

export const HomeLinksSm = () => (
  <React.Fragment>
    <li className="nav-item dropdown sm-dropdown">
      <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <i className="fas fa-align-right" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-plus-square" link="/a/createarticle" label="Add Post" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-plus-square" link="javascript:;" data-toggle="modal" data-target="#search" label="Search" />
        <li>
          <a href="/a/notifications" className="dropdown-item">
            <span className="badge badge-danger">
            5
            </span>
            <i className="fas fa-bell" />
            Notifications
          </a>
        </li>
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
  </React.Fragment>
);

export const NotAuthenticatedLinks = () => (
  <React.Fragment>
    <NavItem classnameOuter="nav-link" classname="fas fa-info-circle" link="/about" label="About Us" />
    <NavItem classnameOuter="nav-link" classname="fas fa-comments" link="/contact" label="Contact Us" />
    <NavButton classnameOuter="nav-link" classname="btn btn-outline-warning btn-sm" link="/" label="Login" />
    <NavButton classnameOuter="nav-link" classname="btn btn-outline-success  btn-sm" link="/register" label="Signup" />
  </React.Fragment>
);

export const NotAuthenticatedLinksSm = () => (
  <React.Fragment>
    <li className="nav-item dropdown sm-dropdown">
      <button
        type="button" className="nav-link dropdown-toggle" id="navbarDropdown" data-toggle="dropdown"
        aria-haspopup="true" aria-expanded="false">
        <i className="fas fa-align-right" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-info-circle" link="/about" label="About Us" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-comments" link="/contact" label="Contact Us" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-in-alt" link="/" label="Login" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-in-alt" link="/register" label="Signup" />
      </ul>
    </li>
  </React.Fragment>
);

export const AuthenticatedLinks = () => (
  <React.Fragment>
    <NavItem classnameOuter="nav-link" classname="fas fa-home" link="/a/home" label="Home" />
    <li className="nav-item dropdown">
      <button type="button" className="nav-link dropdown-toggle btn-link" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <span className="badge badge-danger notifications-badge">
        2
        </span>
        <i className="fas fa-bell" />
        Notifications
      </button>
      <ul className="dropdown-menu" id="dropDownNotifications" aria-labelledby="navbarDropdown">
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
        <DropDownNotification imgSrc={defaultUserIcon} userName="Edward Snowden" message="published an article" period="5 hrs" />
        <div className="dropdown-divider" />
      </ul>
    </li>
    <li className="nav-item dropdown">
      <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <img src={defaultUserIcon} alt="follower avatar" className="followers-avatar navbar-avatar rounded-circle" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
  </React.Fragment>
);

export const AuthenticatedLinksSm = () => (
  <React.Fragment>
    <li className="nav-item dropdown sm-dropdown">
      <button className="nav-link dropdown-toggle" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <i className="fas fa-align-right" />
      </button>
      <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
        <li>
          <a href="/a/notifications" className="dropdown-item">
            <span className="badge badge-danger">
            5
            </span>
            <i className="fas fa-bell" />
            Notifications
          </a>
        </li>
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-user" link="/a/profile" label="Profile" />
        <div className="dropdown-divider" />
        <DropDownItem classnameOuter="dropdown-item" classname="fas fa-sign-out-alt" link="/" label="Logout" />
      </ul>
    </li>
  </React.Fragment>
);
