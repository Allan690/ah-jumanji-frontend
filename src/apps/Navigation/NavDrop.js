import React from 'react';
import PropTypes from 'prop-types';

const NavDrop = (props) => {
  const { num, notifications } = props;

  return (
    <a href="notifications.html" className="dropdown-item">
      <span className="badge badge-danger">{num}</span>
      <i className="fas fa-bell" />
      {notifications}
    </a>
  );
};

const navDropPropTypes = {
  num: PropTypes.string.isRequired,
  notifications: PropTypes.string.isRequired
};

NavDrop.propTypes = navDropPropTypes;

export default NavDrop;
