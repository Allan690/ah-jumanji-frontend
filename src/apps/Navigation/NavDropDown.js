import React from 'react';
import {navPropTypes} from "./NavItem";

const DropDownItem = (props) => {

  const {classnameOuter, classname, link, label} = props;

  return (
    <li>
      <a href={link} className={classnameOuter}>
        <i className={classname} />
        {"  " + label}
      </a>
    </li>
  );
};

DropDownItem.propTypes = navPropTypes;
export default DropDownItem;
