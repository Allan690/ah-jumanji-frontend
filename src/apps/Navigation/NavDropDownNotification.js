import React from 'react';
import PropTypes from 'prop-types';

const DropDownNotification = (props) => {
  const {imgSrc, userName, message, period} = props;

  return (
    <li className="dropdown-item">
      <img src={imgSrc} alt="follower avatar" className="followers-avatar rounded-circle" />
      <button className="btn-link" type="button">
        &nbsp;
        {userName}
      </button>
      &nbsp;
      {message}
      <span className="float-right">
        <button className="btn-sm btn-link" type="button">
          <i className="fas fa-check-circle" />
          Mark As Read
        </button>
      </span>
      <br />
      <i className="text-muted">
        <span className="fas fa-clock " />
        &nbsp;
        {period}
        &nbsp;ago
      </i>
    </li>
  );
};

export const navNotificationPropTypes = {
  imgSrc: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  period: PropTypes.string.isRequired
};

DropDownNotification.propTypes = navNotificationPropTypes;

export default DropDownNotification;
