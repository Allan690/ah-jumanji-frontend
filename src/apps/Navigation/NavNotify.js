import React from 'react';
import PropTypes from 'prop-types';

const NavNotify = (props) => {

  const { src, link, name, notification, markread, time } = props;

  return (
    <li className="dropdown-item">
      <img src={src} alt="follower avatar" className="followers-avatar rounded-circle" />
      <a href={link}>{name}</a>
        {notification}
      <span className="float-right">
        <a href={link} className="btn btn-light btn-sm">
          <i className="fas fa-check-circle" />
          {markread}
        </a>
      </span>
      <br />
      <i className="text-muted">
        <span className="fas fa-clock" />
        {time}
      </i>
    </li>
  );
};

const navNotifyPropTypes = {
  src: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  notification: PropTypes.string.isRequired,
  markread: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired
};

NavNotify.propTypes = navNotifyPropTypes;

export default NavNotify;
