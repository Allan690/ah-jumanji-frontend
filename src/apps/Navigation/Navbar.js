/* eslint-disable react/no-unescaped-entities */
import React from "react";
//import { NavLink } from "react-router-dom";


const Navigation = () => (
  <nav className="navbar navbar-expand-lg navbar-light fixed-top">
    <div className="container">
      <a className="navbar-brand" href="/">
        Author's Haven
      </a>
      <div className="" id="navbarSupportedContent">
        <ul className="navbar-nav ml-auto pull-left">
          <span className="sm-hide">
            <li className="nav-item">
              <a href="/about" className="nav-link">
                <i className="fas fa-info-circle" />
              About Us
              </a>
            </li>
            <li className="nav-item">
              <a href="/contact" className="nav-link">
                <i className="fas fa-comments" />
              Contact Us
              </a>
            </li>
            <li className="nav-item">
              <a href="/" className="btn btn-outline-warning btn-sm">Login</a>
            </li>
            <li className="nav-item">
              <a href="/register" className="btn btn-outline-success  btn-sm">Signup</a>
            </li>
          </span>
          <li className="nav-item dropdown sm-dropdown">
            <button type="button" className="nav-link dropdown-toggle" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
              <i className="fas fa-align-right" />
            </button>
            <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
              <li>
                <a className="dropdown-item" href="/about">
                  <i className="fas fa-info-circle" />
                About Us
                </a>
              </li>
              <li>
                <a href="/contact" className="dropdown-item">
                  <i className="fas fa-comments" />
                Contact Us
                </a>
              </li>
              <div className="dropdown-divider" />
              <li>
                <a className="dropdown-item" href="/">
                  <i className="fas fa-sign-in-alt" />
                Login
                </a>
              </li>
              <li>
                <a className="dropdown-item" href="/register">
                  <i className="fas fa-sign-in-alt" />
                Sign Up
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Navigation;
