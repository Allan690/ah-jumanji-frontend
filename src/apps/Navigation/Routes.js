import React from "react";
import { Switch, Route } from "react-router-dom";
import RegisterContainer from "../Register/components/registerContainer";
import RegisterRedirect from "../Register/components/registerRedirect";
import RegisterRedirectActivated from "../Register/components/registerRedirectActivated";
import SocialAuthentication from '../SocialLogin/components/SocialAuth';
import ForgotPassword from "../resetPassword/components/PasswordResetComponent";
import IndexPage from "../Index/components/IndexPage";
import Rating from "../Rating/components/Rating";
import ResetCodePage from "../resetPassword/components/PasswordResetCodeComponent";
import NotificationHistory from "../Notifiications/components/NotificationHistoryComponent";


const Main = () => (
  <Switch>
    <Route exact path="/" component={IndexPage} />
    <Route exact path="/register" component={RegisterContainer} />
    <Route exact path="/login" component={SocialAuthentication} />
    <Route exact path="/register-redirect" component={RegisterRedirect} />
    <Route exact path="/activate-account/:uid/:token" component={RegisterRedirectActivated} />
    <Route exact path="/reset_password" component={ForgotPassword} />
    <Route exact path="/reset_code/:reset_code" component={ResetCodePage} />
    <Route exact path="/rating" component={Rating} />
    <Route exact path="/register" component={RegisterContainer} />
    <Route exact path="/login" component={SocialAuthentication} />
    <Route exact path="/register-redirect" component={RegisterRedirect} />
    <Route exact path="/activate-account/:uid/:token" component={RegisterRedirectActivated} />
    <Route exact path="/reset_password" component={ForgotPassword} />
    <Route exact path="/reset_code/:reset_code" component={ResetCodePage} />
    <Route exact path="/a/notifications" component={NotificationHistory} />
  </Switch>
);

export default Main;
