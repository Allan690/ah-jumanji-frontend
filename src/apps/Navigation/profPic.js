import React from 'react';
import config from '../../config/config';


class ProfPic extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      isLoading: true,
    };
  }

  componentDidMount () {
    setTimeout(() => {
      this.setState({ isLoading: false });
    }, 3600);
  }

  render() {
    let {isLoading} = this.state;
    let avatar = config.avatarImage.avatarUrl;
    return (
      <button className="nav-link dropdown-toggle btn-link" type="button" id="navbarDropdown" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
        <img src={isLoading ? avatar : localStorage.profPic} alt="follower avatar" className="followers-avatar navbar-avatar rounded-circle" />
      </button>
    );
  }
}

export default ProfPic;
