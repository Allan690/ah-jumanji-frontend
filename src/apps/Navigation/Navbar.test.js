import React from 'react';

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import Navigation from './Navbar';
import ProfPic from './profPic';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    location: {pathname: "/a/home"}
  };

  const mountedEnzymeWrapper = shallow(<Navigation {...props} />);

  return {
    props,
    mountedEnzymeWrapper
  };
}

describe('Navigation', () => {
  it('should render self and subcomponents', () => {
    const { mountedEnzymeWrapper } = setup();

    expect(mountedEnzymeWrapper.find('nav').hasClass('navbar-expand-lg')).toBe(true);
  });
});

describe('ProfPic: ', () => {
  let profPic;
  beforeEach(() => {
    profPic = shallow(<ProfPic />);
  });

  it('gets profile photo: ', () => expect(profPic.exists()).toEqual(true));

});
