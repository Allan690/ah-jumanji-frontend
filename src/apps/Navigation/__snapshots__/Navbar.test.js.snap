// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly in "debug" mode 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light fixed-top"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Author's Haven
    </a>
    <div
      className=""
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav ml-auto pull-left"
      >
        <span
          className="sm-hide"
        >
          <NavItem
            classname="fas fa-info-circle"
            classnameOuter="nav-link"
            label="About Us"
            link="/about"
          />
          <NavItem
            classname="fas fa-comments"
            classnameOuter="nav-link"
            label="Contact Us"
            link="/contact"
          />
          <NavButton
            classname="btn btn-outline-warning btn-sm"
            classnameOuter="nav-link"
            label="Login"
            link="/"
          />
          <NavButton
            classname="btn btn-outline-success  btn-sm"
            classnameOuter="nav-link"
            label="Signup"
            link="/register"
          />
          <NotificationsBadge
            notifications_count={5}
          />
        </span>
        <li
          className="nav-item dropdown sm-dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            id="navbarDropdown"
            type="button"
          >
            <i
              className="fas fa-align-right"
            />
          </button>
          <ul
            aria-labelledby="navbarDropdown"
            className="dropdown-menu"
          >
            <DropDownItem
              classname="fas fa-info-circle"
              classnameOuter="dropdown-item"
              label="About Us"
              link="/about"
            />
            <DropDownItem
              classname="fas fa-comments"
              classnameOuter="dropdown-item"
              label="Contact Us"
              link="/contact"
            />
            <div
              className="dropdown-divider"
            />
            <DropDownItem
              classname="fas fa-sign-in-alt"
              classnameOuter="dropdown-item"
              label="Login"
              link="/"
            />
            <DropDownItem
              classname="fas fa-sign-in-alt"
              classnameOuter="dropdown-item"
              label="Signup"
              link="/register"
            />
          </ul>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
